src/name_core.jl

```diff
@@ -0,0 +1,97 @@
+
+"""
+    name2dim(dimnames, [name])
+
+For `dimnames` being a tuple of dimnames (symbols) for dimenensions.
```

Comment:
I might go with `name2dim(dn, [name])` to avoid having a function called `dim_names` and a variable called `dimnames`.

Reply:
Perhaps `dim_labels`?

Reply:
Sure... or maybe `name2dim(names, [labels])`?

Reply:
now that it is called `dim` this is resolved